blocks/form/form.js

```diff
@@ -0,0 +1,213 @@
+  const heading = document.createElement('h3');
+  heading.textContent = fd.Label;
+  return heading;
+}
+
+function createInput(fd) {
+  const input = document.createElement('input');
+  input.type = fd.Type;
+  input.id = fd.Field;
+  input.name = fd.Field;
+  input.value = fd.Value || '';
+  if (fd.Placeholder) {
+    input.setAttribute('placeholder', fd.Placeholder);
+  }
+  if (fd.Mandatory === 'x') {
+    input.setAttribute('required', 'required');
+  }
+  return input;
+}
+
+function createTextArea(fd) {
+  const input = document.createElement('textarea');
+  input.name = fd.Field;
+  input.value = fd.Value || '';
+  if (fd.Placeholder) {
+    input.setAttribute('placeholder', fd.Placeholder);
+  }
+  if (fd.Mandatory === 'x') {
+    input.setAttribute('required', 'required');
+  }
+  return input;
+}
+
+function createSelect(fd) {
+  const select = document.createElement('select');
+  select.name = fd.Field;
+  select.value = fd.Value || '';
+  if (fd.Placeholder) {
+    const ph = document.createElement('option');
+    ph.textContent = fd.Placeholder;
+    ph.setAttribute('selected', '');
+    ph.setAttribute('disabled', '');
+    select.append(ph);
+  }
+  fd.Options.split(',').forEach((o) => {
+    const option = document.createElement('option');
+    option.textContent = o.trim();
+    option.value = o.trim();
+    select.append(option);
+  });
+  if (fd.Mandatory === 'x') {
+    select.setAttribute('required', 'required');
+  }
+  return select;
+}
+
+function createLabel(fd) {
+  const label = document.createElement('label');
+  label.setAttribute('for', fd.Field);
+  label.textContent = fd.Label;
+  if (fd.Mandatory === 'x') {
+    label.classList.add('required');
+  }
+  if (fd.Extra) {
+    const [linkText, href] = fd.Extra.split(',').map((str) => str.trim());
+    if (href) {
+      // insert link in label text
+      const labelTexts = label.textContent.split(linkText);
+      label.innerHTML = '';
+
+      const link = document.createElement('a');
+      link.textContent = linkText;
+      link.href = href;
+      link.target = '_blank';
+
+      label.append(labelTexts[0]);
+      label.append(link);
+      label.append(labelTexts[1]);
+    }
+  }
+  return label;
+}
+
+function applyRules(form, rules) {
+  const payload = constructPayload(form);
+  rules.forEach((field) => {
+    const { type, condition: { key, operator, value } } = field.rule;
+    if (type === 'visible') {
+      if (operator === 'eq') {
+        if (payload[key] === value) {
+          form.querySelector(`.${field.fieldId}`).classList.remove('hidden');
+        } else {
+          form.querySelector(`.${field.fieldId}`).classList.add('hidden');
+        }
+      }
+    }
+  });
+}
+
+async function createForm(formURL) {
+  const { pathname } = new URL(formURL);
+  const resp = await fetch(pathname);
+  const json = await resp.json();
+  const form = document.createElement('form');
+  const rules = [];
+  // eslint-disable-next-line prefer-destructuring
```

Comment:
couldn't we just destruct here if eslint recommends it?

Reply:
@nc-andreashaller https://github.com/hlxsites/netcentric/pull/82